core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProvider.scala

```diff
@@ -249,28 +273,64 @@ class DataRepoEntityProvider(snapshotModel: SnapshotModel, requestArguments: Ent
     }
   }
 
-  private def runBigQueryQueries(entityNameColumn: String, bqQueryJobConfigs: Map[Set[ParsedEntityLookupExpression], QueryJobConfiguration], petKey: String): IO[List[ExpressionAndResult]] = {
-    bqServiceFactory.getServiceForPet(petKey).use { bqService =>
-      for {
-        queryResults <- bqQueryJobConfigs.toList.traverse { // consider parTraverse if serial is slow but be careful not to use too many threads
-          case (expressions, bqJob) => bqService.query(bqJob).map(expressions -> _)
-        }
-        _ <- checkQuerySize(queryResults)
-      } yield {
-        transformQueryResultToExpressionAndResult(entityNameColumn, queryResults)
-      }
-    }
+  private def runBigQuery(bqQueryJobConfigBuilder: QueryJobConfiguration.Builder, petKey: String): IO[TableResult] = {
+    bqServiceFactory.getServiceForPet(petKey).use(_.query(
+      bqQueryJobConfigBuilder
+        .setMaximumBytesBilled(config.bigQueryMaximumBytesBilled)
+        .build()))
   }
 
-  private def transformQueryResultToExpressionAndResult(entityNameColumn: String, queryResults: List[(Set[ParsedEntityLookupExpression], TableResult)]): List[ExpressionAndResult] = {
+  /**
+    * The tableResult should have a single row per root entity. Each row should have a column for each
```

Comment:
This first sentence makes a lot of sense now that I understand things more.  Can we use this to our advantage is estimating sizes of result sets?  For example, in the simplest case, if the root table has more rows than our threshold memory size, we can preemptively alert the user.  Smarter thing would be to do some rough calculation based on maybe the number of unique expressions and the smallest data type size, that sort of thing.  I guess it would boil down to asking whether it would be worthwhile to preemptively alert users that their queries may fail based on the fact that their root table is so large.

Reply:
we already do this, see the function `checkSubmissionSize`